automation_tools/satellite6/upgrade/__init__.py

```diff
@@ -166,31 +166,44 @@ def product_upgrade(product):
         to_version = os.environ.get('TO_VERSION')
         sat_host, cap_hosts, clients6, clients7 = setup_products_for_upgrade(
             product, os.environ.get('OS'))
-        with LogAnalyzer(sat_host):
-            if from_version != to_version:
-                current_version = execute(
-                    get_sat_version, host=sat_host)
-                if not current_version == to_version:
-                    execute(satellite6_upgrade, host=sat_host)
-            elif from_version == to_version:
-                execute(satellite6_zstream_upgrade, host=sat_host)
-            # Generate foreman debug on satellite after upgrade
+        try:
+            with LogAnalyzer(sat_host):
+                if from_version != to_version:
+                    current_version = execute(
+                        get_sat_version, host=sat_host)
+                    if not current_version == to_version:
+                        execute(satellite6_upgrade, host=sat_host)
+                elif from_version == to_version:
```

Comment:
This can safely be a `else` block since the first condition is that `from_version` != `to_version`